Hippo.Core/Services/AccountUpdateService.cs

```diff
@@ -241,6 +242,37 @@ private async Task<Result> CompleteUpdateSshKey(QueuedEventDataModel data)
             // _dbContext.SaveAsync() is handled elsewhere for this change
             return Result.Ok();
         }
+
+        private async Task<Result> CompleteRemoveGroupMember(QueuedEventDataModel data)
+        {
+            var accountModel = data.Accounts.FirstOrDefault();
+            var groupModel = data.Groups.FirstOrDefault();
+            var account = await _dbContext.Accounts
+                .Include(a => a.MemberOfGroups.Where(g => g.Name == groupModel.Name))
+                .Where(a => a.Cluster.Name == data.Cluster && a.Owner.Kerberos == accountModel.Kerberos)
+                .FirstOrDefaultAsync();
+
+            if (accountModel == null || groupModel == null)
+            {
+                return Result.Error("Invalid data: action {Action} requires one account and one group", QueuedEvent.Actions.AddAccountToGroup);
+            }
+
+            if (account == null)
+            {
+                return Result.Error("Account not found: {Kerberos} on cluster {Cluster}", accountModel.Kerberos, data.Cluster);
+            }
+
+            if (!account.MemberOfGroups.Any())
+            {
+                return Result.Error("Account {Kerberos} not a member of group {Name} on cluster {Cluster}", account.Kerberos, groupModel.Name, data.Cluster);
+            }
+
+            // We're only removing the one group that was included...
+            account.MemberOfGroups.Clear();
```

Comment:
i'm kinda surprised this works but if you tested and it looks good, then i learned something new.